server/security.py

```
@@ -0,0 +1,210 @@
+            return proc.returncode or 1, out, err
+        return proc.returncode or 0, out, err
+    except asyncio.TimeoutError:
+        return -1, "", f"Command timed out after {timeout}s"
+
+
+def chpasswd_entry(username: str, password: str) -> tuple[int, str, str]:
+    """Set password via chpasswd stdin (no shell interpolation)."""
+    base = username[6:] if username.startswith("dnstt_") else username
+    validate_vpn_username(base)
+    proc = subprocess.run(
+        ["sudo", "chpasswd"],
```

Ruff (0.15.14)

[error] Starting a process with a partial executable path

(S607)